app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java

```diff
@@ -1,18 +1,200 @@
+				showDialog(DATE_DIALOG_ID);
+				return false;
+			}
+		});
+
+		Preference btnTimeFilter = findPreference("btnTimeFilter");
+		btnTimeFilter.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+			@TargetApi(Build.VERSION_CODES.HONEYCOMB)
+			@Override
+			public boolean onPreferenceClick(Preference preference) {
+				showDialog(TIME_DIALOG_ID);
+				return false;
+			}
+		});
+
+		JodaTimeAndroid.init(this);
+
+		// Get saved user settings
+		SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(contextOfApplication);
+
+		// The month in the date picker is in 0-11 range
+		expirationMonth = settings.getInt("Month", 1) - 1;
+		expirationDay = settings.getInt("Day", 1);
+		expirationHour = settings.getInt("Hour", 23);
+		expirationMinute = settings.getInt("Minute", 30);
+		isLastDay = settings.getBoolean("LastDay", false);
+		expirationYear = settings.getInt("Year", 2016);
+
+		initSummary(getPreferenceScreen());
+	}
+
+	@Override
+	protected void onPostCreate(Bundle savedInstanceState) {
+		super.onPostCreate(savedInstanceState);
+		getDelegate().onPostCreate(savedInstanceState);
+	}
+
+	public ActionBar getSupportActionBar() {
+		return getDelegate().getSupportActionBar();
+	}
+
+	public void setSupportActionBar(@Nullable Toolbar toolbar) {
+		getDelegate().setSupportActionBar(toolbar);
+	}
+
+	@Override
+	public MenuInflater getMenuInflater() {
+		return getDelegate().getMenuInflater();
+	}
+
+	@Override
+	public void setContentView(@LayoutRes int layoutResID) {
+		getDelegate().setContentView(layoutResID);
+	}
+
+	@Override
+	public void setContentView(View view) {
+		getDelegate().setContentView(view);
+	}
+
+	@Override
+	public void setContentView(View view, ViewGroup.LayoutParams params) {
+		getDelegate().setContentView(view, params);
+	}
+
+	@Override
+	public void addContentView(View view, ViewGroup.LayoutParams params) {
+		getDelegate().addContentView(view, params);
+	}
+
+	@Override
+	protected void onPostResume() {
+		super.onPostResume();
+		getDelegate().onPostResume();
+	}
+
+	@Override
+	protected void onTitleChanged(CharSequence title, int color) {
+		super.onTitleChanged(title, color);
+		getDelegate().setTitle(title);
+	}
+
+	@Override
+	public void onConfigurationChanged(Configuration newConfig) {
+		super.onConfigurationChanged(newConfig);
+		getDelegate().onConfigurationChanged(newConfig);
+	}
+
+	@Override
+	protected void onStop() {
+		super.onStop();
+		getDelegate().onStop();
+		saveSettings();
+	}
+
+	@Override
+	protected void onDestroy() {
+		super.onDestroy();
+		getDelegate().onDestroy();
+	}
+
+	@Override
+	protected void onResume() {
+		super.onResume();
+		// Set up a listener whenever a key changes
+		getPreferenceScreen().getSharedPreferences()
+			.registerOnSharedPreferenceChangeListener(this);
+	}
+
+	@Override
+	protected void onPause() {
+		super.onPause();
+		// Unregister the listener whenever a key changes
+		getPreferenceScreen().getSharedPreferences()
```

Comment:
I don't know if you are following an style guide of coding but 

the more comfortable way to indent this kind of things I think is:

```
getPreferenceScreen()
    .getSharedPreferences()
    .unregisterOnSharedPreferenceChangeListener(this);
```

or

```
getPreferenceScreen()
.getSharedPreferences()
.unregisterOnSharedPreferenceChangeListener(this);
```